fix(DB/SAI): Duskwood Flesh Eating Worms

LuckyLuc96 opened the pull request:
Update Flesh Eating Worms in Duskwood to attack on spawn. Previously, they would ignore users in melee range when they spawn in, and the field for AIName was empty.

<!-- First of all, THANK YOU for your contribution. -->

## Changes Proposed:
<!-- If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address. -->
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:
<!-- If your fix has a relating issue, link it below -->
- Closes https://github.com/chromiecraft/chromiecraft/issues/8055

## SOURCE:
<!-- If you can, include a source that can strengthen your claim -->
The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- See [here](https://www.youtube.com/watch?v=3SJXrvkP_o8) at 1:03 and 2:00
- Also see [here](https://www.wowhead.com/wotlk/spell=3428/summon-flesh-eating-worms) where summoned unit's lifespan is given as 30 seconds.
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:
<!-- Does it build without errors? Did you test in-game? What did you test? On which OS did you test? Describe any other tests performed -->
This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.


## How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. .tele ravenhillcemetery
2. Kill a Rotted One from melee range
3. See the the spawned creature will attack from melee range, which previously it would not.

## Known Issues and TODO List:
<!-- Is there anything else left to do after this PR? -->

- [ ]
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/GyFvXpk7)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

Nyeriah commented:
I don't think this AI is used. The issue is likely that grid notifiers don't update unless the player moves, so it doesn't attack on spawn

LuckyLuc96 commented:
Do you think changing it to the warrior npc standard AI would suffice? I am
unfamiliar with grid notifiers. I can look into it

On Fri, May 30, 2025, 9:56 AM Andrew ***@***.***> wrote:

> *Nyeriah* left a comment (azerothcore/azerothcore-wotlk#22189)
> <https://github.com/azerothcore/azerothcore-wotlk/pull/22189#issuecomment-2922470174>
>
> I don't think this AI is used. The issue is likely that grid notifiers
> don't update unless the player moves, so it doesn't attack on spawn
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/azerothcore/azerothcore-wotlk/pull/22189#issuecomment-2922470174>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWOVFQIM6WW4N7DSKYLDVNL3BBPRHAVCNFSM6AAAAAB5ZOCP4SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSMRSGQ3TAMJXGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

LuckyLuc96 commented:
I added this source to the PR body, but basically I added logic where once the summoner unit dies (the Rotted One) and the worm spawns, the worm will attack any player within 18 yards. This distance is the same as it's set "vision" distance in it's mob template data. 

Also on researching this mob, I noticed that it's supposed to have a limited life span of 30 seconds. The 2nd part of the smart script will ensure that the worm will die after 30 seconds.

LuckyLuc96 commented:
The worms are ready to be unleashed into the world

sudlud commented:
Please use keira3 to generate comments foe the sai entries. Also please update the pr title

LuckyLuc96 commented:
> thanks for the update of the comments. but i can really only suggest to look into using Keira3 for SAI edits and comment generation, this helps keeping a standard for the comments. Did you write the comments yourself or are you maybe using a different editor here? :)

My apologies. I was not aware of the auto-generated comment feature for smart ai in keira.
